pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -409,27 +409,28 @@ func (m *manager) canAdmitPod(allocatedPods []*v1.Pod, pod *v1.Pod) (bool, strin
 func (m *manager) canResizePod(allocatedPods []*v1.Pod, pod *v1.Pod) (bool, string, string) {
 	// TODO: Move this logic into a PodAdmitHandler by introducing an operation field to
 	// lifecycle.PodAdmitAttributes, and combine canResizePod with canAdmitPod.
+	cpuRequests := resource.GetResourceRequest(pod, v1.ResourceCPU)
+	memRequests := resource.GetResourceRequest(pod, v1.ResourceMemory)
 	if v1qos.GetPodQOS(pod) == v1.PodQOSGuaranteed && !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScalingExclusiveCPUs) {
-		if m.containerManager.GetNodeConfig().CPUManagerPolicy == "static" {
+		if m.containerManager.GetNodeConfig().CPUManagerPolicy == "static" &&
+			(cpuRequests != resource.GetResourceAllocated(pod, v1.ResourceCPU)) {
```

Comment:
When would we hit this case? If `desired == allocated`, no resize action needs to be taken. I think we would hit the `!updated` case in `HandlePodResourcesResize` and shouldn't end up calling `canResizePod`

Reply:
Oh, I just realized we could hit this case if one container is sized up but another is sized down, but the aggregate cpu requests across all containers is the same. Was that the intention?

Reply:
Oh, I just realized we will hit this case if memory is being resized but CPU isn't. My bad!

Reply:
Since `cpuRequests` and `resource.GetResourceAllocated(pod, v1.ResourceCPU)` look to be an aggregate total across all containers, is it possible we would hit this case if one container has its CPU sized up and another has its CPU sized down (such that the total for the pod is the same)? In such a case, we would still want to mark the resize as infeasible, right?

Reply:
Thanks. You are right that looking at aggregates might miss some corner cases. We would need to look at individual containers to decide.

Reply:
I've added a check that loops through containers to compare requested and actuated resources. However, I am thinking if there could be any edge case where request could equal actuated during resize, which would prevent Guaranteed QoS pods from being filtered out.